lib/drivers/i-sens/careSens.js

```diff
@@ -172,13 +327,142 @@ class CareSens {
       CareSens.verifyChecksum(bytes, response.crc);
     }
 
-    return response;
+    const err = common.getName(ERROR, response.command);
+    if (err !== 'unknown') {
+      throw new Error(err);
+    } else {
+      return response;
+    }
+  }
+
+  static extractPacket(bytes) {
+    const packet = {
+      bytes,
+      valid: false,
+      packet_len: 0,
+      payload: null,
+    };
+
+    const packetLength = bytes.length;
+    packet.packet_len = packetLength;
+    packet.valid = true;
```

Comment:
Not strictly necessary to change, but couldn't these assignments be done in the initial definition at 339?

Reply:
:+1: